compiler/src/iree/compiler/Codegen/ExternalInterfaces/GPUEncodingExternalModels.cpp

```diff
@@ -337,8 +368,40 @@ struct GPUDeviceEncodingLayoutAttrInterface
     if (!linalgOp) {
       return nullptr;
     }
+    DictionaryAttr config = layoutAttr.getConfiguration();
+    if (!config) {
+      return nullptr;
+    }
+    IREE::GPU::TargetAttr gpuAttr = getGPUTargetNamedAttr(config);
+    if (!gpuAttr) {
+      return nullptr;
+    }
```

Comment:
We do not need the check for `DictionaryAttr` if you move the logic to `getGPUTargetAttr`. If the attribute is nullptr, the method returns nullptr.